api/src/main/java/org/openmrs/module/locationbasedaccess/utils/LocationUtils.java

```diff
@@ -36,47 +38,52 @@ public static Location getPersonLocation(Person person) {
         return null;
     }
 
-    public static Boolean doesPersonBelongToGivenLocation(Person person, PersonAttributeType personAttributeType, String sessionLocationUuid) {
+    public static Boolean doesPersonBelongToGivenLocations(Person person, PersonAttributeType personAttributeType, List<String> sessionLocationsLists) {
         PersonAttribute personAttribute = person.getAttribute(personAttributeType);
-        return (personAttribute != null && compare(personAttribute.getValue(), sessionLocationUuid));
+        return (personAttribute != null && sessionLocationsLists.contains(personAttribute.getValue()));
     }
 
-    public static Boolean doesUserBelongToGivenLocation(User user, String sessionLocationUuid) {
-        String userLocationProperty = user.getUserProperty(LocationBasedAccessConstants.LOCATION_USER_PROPERTY_NAME);
-        return (userLocationProperty != null && compare(userLocationProperty, sessionLocationUuid));
+    public static Boolean doesUserBelongToGivenLocations(User user, List<String> sessionLocationsLists){
+        List<String> accessibleLocationsList = getUserAccessibleLocationUuids(user);
+        if(accessibleLocationsList != null ){
+            for(String location : accessibleLocationsList ){
+                if(!StringUtils.isBlank(location) && sessionLocationsLists.contains(location)){
+                    return true;
+                }
+            }
+        }
+        return false ;
     }
 
-    public static Boolean doesUsersForPersonBelongToGivenLocation(Person person, String sessionLocationUuid) {
+    public static Boolean doesUsersForPersonBelongToGivenLocations(Person person, List<String> sessionLocationsLists) {
         List<User> userList = Context.getUserService().getUsersByPerson(person, false);
         for (Iterator<User> iterator = userList.iterator(); iterator.hasNext(); ) {
-            if(LocationUtils.doesUserBelongToGivenLocation(iterator.next(), sessionLocationUuid)) {
+            if(LocationUtils.doesUserBelongToGivenLocations(iterator.next(), sessionLocationsLists)) {
                 return true;
             }
         }
         return false;
     }
 
-    public static Boolean compare(String value1, String value2) {
-        return (StringUtils.isNotBlank(value1) && StringUtils.isNotBlank(value2)) && value1.equals(value2);
-    }
-
     /**
      * Used to get the accessible location for the user. It will first get from the user property. if the user
```

Comment:
If you change a method, then you need to think about the Java docs also

Reply:
Have you seen this @VANKINEENITAWRUN ?

Reply:
yes fixed it